pkg/controller/framework/shared_informer.go

```diff
@@ -48,42 +48,48 @@ type SharedInformer interface {
 
 type SharedIndexInformer interface {
 	SharedInformer
-
-	AddIndexer(indexer cache.Indexer) error
+	AddIndexers(indexers cache.Indexers) error
 	GetIndexer() cache.Indexer
 }
 
 // NewSharedInformer creates a new instance for the listwatcher.
 // TODO: create a cache/factory of these at a higher level for the list all, watch all of a given resource that can
 // be shared amongst all consumers.
 func NewSharedInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPeriod time.Duration) SharedInformer {
-	sharedInformer := &sharedInformer{
-		processor: &sharedProcessor{},
-		store:     cache.NewStore(DeletionHandlingMetaNamespaceKeyFunc),
+	sharedInformer := &sharedIndexInformer{
+		processor:        &sharedProcessor{},
+		indexer:          cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, cache.Indexers{}),
+		listerWatcher:    lw,
+		objectType:       objType,
+		fullResyncPeriod: resyncPeriod,
 	}
+	return sharedInformer
+}
 
-	fifo := cache.NewDeltaFIFO(cache.MetaNamespaceKeyFunc, nil, sharedInformer.store)
-
-	cfg := &Config{
-		Queue:            fifo,
-		ListerWatcher:    lw,
-		ObjectType:       objType,
-		FullResyncPeriod: resyncPeriod,
-		RetryOnError:     false,
-
-		Process: sharedInformer.HandleDeltas,
+// NewSharedIndexInformer creates a new instance for the listwatcher.
+// TODO: create a cache/factory of these at a higher level for the list all, watch all of a given resource that can
+// be shared amongst all consumers.
+func NewSharedIndexInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPeriod time.Duration, indexers cache.Indexers) SharedIndexInformer {
+	sharedIndexInformer := &sharedIndexInformer{
+		processor:        &sharedProcessor{},
+		indexer:          cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers),
+		listerWatcher:    lw,
+		objectType:       objType,
+		fullResyncPeriod: resyncPeriod,
 	}
-	sharedInformer.controller = New(cfg)
```

Comment:
@wojtek-t you can't create the controller here, because the indexer this pull leaves indexers as immutable (no change to indexes), which makes the easy to reason about, but it means that controller created here wouldn't have the correct indexes.

I think that either way, creating the controller at the point of use will be sufficient for what we're trying to accomplish since there's actually no linkage between this controller and the one returned by the method that gives back a fake interface.